google/web-stories-wp
# Dashboard: FiltersProvider optimizations incorporate Authors filter

## Context


Consolidate dashboard filters into one provider.  Optimize FiltersProvider and incorporate the authors filter.

## Summary


Adds author filtering functionality to the FiltersProvider. Refactor existing use of authors filter. Update `datalist` component.

## Relevant Technical Choices


The filters now allow for caching and postponing primary option fetching with `getPrimaryOptions`.  This allows other components, like the `datalist`, to build UI with known initial data and update the `options` once they've been fetched or pulled from a cache.

## To-do



## User-facing changes



## Testing Instructions






This PR can be tested by following these steps:

1. Go to Story Dashboard
2. Use filters
 - All taxonomy filters, and the Authors filter have changed
3. Check the Story Editor for any DropDown issues
 - The only one I noticed that was affected is Story Editor -> Document -> Publishing -> Author

## Reviews

### Does this PR have a security-related impact?
No


### Does this PR change what data or activity we track or use?
No


### Does this PR have a legal-related impact?
No


## Checklist




---



Fixes #11620
